v8.0 - IdentityServer (#1081)
DuendeSoftware/docs.duendesoftware.com: astro/src/content/docs/identityserver/reference/v7/dcr/validation.md

|-------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `ValidateSoftwareStatementAsync(…)` | Validates the software statement of the request. The default implementation does nothing, and is included as an extension point.                                                                                                                                                                                |
| `SetGrantTypesAsync(…)`             | Validates requested grant types and uses them to set the allowed grant types of the client.                                                                                                                                                                                                                     |

> | `SetRedirectUrisAsync(…)`           | Validates requested redirect uris and uses them to set the redirect uris of the client.                                                                                                                                                                                                                         |

Reviewer: ```suggestion
| `SetRedirectUrisAsync(…)`           | Validates requested redirect URIs and uses them to set the redirect URIs of the client.                                                                                                                                                                                                                         |
```

| `SetScopesAsync(…)`                 | Validates requested scopes and uses them to set the scopes of the client.                                                                                                                                                                                                                                       |
| `SetDefaultScopes(…)`               | Sets scopes on the client when no scopes are requested. The default implementation sets no scopes and is intended as an extension point.                                                                                                                                                                        |

> | `SetSecretsAsync(…)`                | Validates the requested jwks to set the secrets of the client.                                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| `SetSecretsAsync(…)`                | Validates the requested JSON Web Key set to set the secrets of the client.                                                                                                                                                                                                                                                  |
```

| `SetClientNameAsync(…)`             | Validates the requested client name uses it to set the name of the client.                                                                                                                                                                                                                                      |

> | `SetLogoutParametersAsync(…)`       | Validates the requested client parameters related to logout and uses them to set the corresponding properties in the client. Those parameters include the post logout redirect uris, front channel and back channel uris, and flags for the front and back channel uris indicating if they require session ids. |

Reviewer: ```suggestion
| `SetLogoutParametersAsync(…)`       | Validates the requested client parameters related to logout and uses them to set the corresponding properties in the client. Those parameters include the post logout redirect URIs, front channel and back channel URIs, and flags for the front and back channel URIs indicating if they require session ids. |
```

> | `SetMaxAgeAsync(…)`                 | Validates the requested default max age and uses it to set the user sso lifetime of the client.                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| `SetMaxAgeAsync(…)`                 | Validates the requested default max age and uses it to set the user SSO lifetime of the client.                                                                                                                                                                                                                 |
```

> | `SetUserInterfaceProperties(…)`     | Validates details of the request that control the user interface, including the logo uri, client uri, initiate login uri, enable local login flag, and identity provider restrictions, and uses them to set the corresponding client properties.                                                                |

Reviewer: ```suggestion
| `SetUserInterfaceProperties(…)`     | Validates details of the request that control the user interface, including the logo URI, client URI, initiate login URI, enable local login flag, and identity provider restrictions, and uses them to set the corresponding client properties.                                                                |
```

> | `SetPublicClientProperties(…)`      | Validates the requested client parameters related to public clients and uses them to set the corresponding properties in the client. Those parameters include the require client secret flag and the allowed cors origins.                                                                                      |

Reviewer: ```suggestion
| `SetPublicClientProperties(…)`      | Validates the requested client parameters related to public clients and uses them to set the corresponding properties in the client. Those parameters include the require client secret flag and the allowed CORS origins.                                                                                      |
```
